fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/ExtractCommonFactorRule.java

```diff
@@ -144,22 +138,6 @@ private static Expression extractCommonFactors(CompoundPredicate originPredicate
         LinkedHashMap<Set<Integer>, Set<Expression>> commonFactorPartitions
                 = partitionByMostCommonFactors(commonFactorToPartIds);
 
-        int extractedExpressionNum = 0;
-        for (Set<Expression> exprs : commonFactorPartitions.values()) {
-            extractedExpressionNum += exprs.size();
-        }
-
-        // no any common factor
-        if (commonFactorPartitions.entrySet().iterator().next().getKey().size() <= 1
-                && !(originPredicate.getWidth() > leftDeapTreePredicate.getWidth())
-                && originExpressionNum <= extractedExpressionNum) {
-            // this condition is important because it can avoid deap loop:
-            // origin originExpr:               A = 1 and (B > 0 and B < 10)
-            // after ExtractCommonFactorRule:   (A = 1 and B > 0) and (B < 10)     (left deap tree)
-            // after SimplifyRange:             A = 1 and (B > 0 and B < 10)       (right deap tree)
-            return originPredicate;
-        }
-
         // now we can do extract common factors for each part:
```

Comment:
how can we infer extractedExprs from originPredicate, initPartitions and commonFactorPartitions